tf_encrypted/keras/testing_utils.py

```diff
@@ -0,0 +1,109 @@
+"""Testing utilities for tfe.keras."""
+import numpy as np
+import tensorflow as tf
+
+import tf_encrypted as tfe
+
+
+def agreement_test(layer_cls, kwargs=None, input_shape=None,
+                   input_data=None, rtol=1e-3, **tfe_kwargs):
+  """Check agreement between a tf.keras layer and a tfe.keras layer.
+  Arguments:
+    layer_cls: Layer class object (from tfe.keras).
+    kwargs: Optional dictionary of keyword arguments for instantiating the
+      layers.
+    input_shape: Input shape tuple.
+    input_data: Numpy array of input data.
+    expected_output: Shape tuple for the expected shape of the output.
+    tfe_kwargs: Additional kwargs to supply the tfe.keras Layer not included in
+      the argspec of the original tf.keras Layer object.
+  Raises:
+    ValueError: if `input_data is None and input_shape is None`.
+  """
+  input_shape, input_data = _sanitize_testing_args(input_shape, input_data)
+
+  tf_layer_cls = getattr(tf.keras.layers, layer_cls.__name__)
+  tfe_kwargs = {**kwargs, **tfe_kwargs}
+
+  with tfe.protocol.SecureNN() as prot:
+    tfe_layer = layer_cls(**tfe_kwargs)
+    x = prot.define_private_variable(input_data)
```

Comment:
you should be able to call `tfe.define_private_variable` and avoid the explicit protocol reference